src/Data/Fixed.purs

```diff
@@ -271,10 +306,134 @@ approxDiv a b = Fixed (x * n / y)
     y = numerator b
     n = BigInt.fromInt (denominator a)
 
+-- | Parse a fixed-precision number from a string. Any decimal digits which are
+-- | not representable in the specified precision will be ignored.
+-- |
+-- | ```
+-- | > fromString "123.456" :: Maybe (Fixed P1000)
+-- | (Just (fromString "123.456" :: P1000))
+-- | ```
+-- |
+-- | Where possible, this function should be preferred over `fromNumber`, since
+-- | it is exact (whereas `fromNumber` can only provide an approximation for
+-- | larger inputs).
+-- |
+-- | ```
+-- | > fromString "9007199254740992.5" :: Maybe (Fixed P10)
+-- | (Just (fromString "9007199254740992.5" :: P10))
+-- |
+-- | > fromNumber 9007199254740992.5 :: Maybe (Fixed P10)
+-- | (Just (fromString "9007199254740992.0" :: P10))
+-- | ```
+-- |
+fromString
+  :: forall precision
+   . KnownPrecision precision
+  => String
+  -> Maybe (Fixed precision)
+fromString str =
+  let
+    numDigits = reflectPrecisionDecimalPlaces (PProxy :: PProxy precision)
+    denom = BigInt.fromInt (reflectPrecision (PProxy :: PProxy precision))
+
+    isDigit = between '0' '9'
+    wholeDigits = StringCU.countPrefix isDigit str
+    { before, after } = StringCU.splitAt wholeDigits str
```

Comment:
Why not use `takeWhile`?

Reply:
Because I want to get the bit that's left over, too. I could also use `takeWhile`, measure the length of the the prefix I get, and then drop that many characters, but that doesn't seem very different from this approach to me?